test/integration/collection-management/collection.test.ts

```diff
@@ -95,18 +95,14 @@ describe('Collection', function () {
       ]);
     });
 
-    it('should fail due to illegal listCollections', function (done) {
-      expect(() => db.collection(5)).to.throw('Collection name must be a String');
-      expect(() => db.collection('')).to.throw('Collection names cannot be empty');
-      expect(() => db.collection('te$t')).to.throw("Collection names must not contain '$'");
-      expect(() => db.collection('.test')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test.')).to.throw(
-        "Collection names must not start or end with '.'"
-      );
-      expect(() => db.collection('test..t')).to.throw('Collection names cannot be empty');
-      done();
+    it('should fail on server due to illegal collection name', async function () {
+      try {
+        const illegalCollection = await db.createCollection('a\x00b');
+        await illegalCollection.insertOne({ a: 1 });
+        expect.fail('a MongoServerError was expected due to illegal collection name');
+      } catch (error) {
+        expect(error instanceof MongoServerError);
```

Comment:
chai assertion

Reply:
I'll fix the assertions for this one, too!